scripts/build.sh

```diff
@@ -1,5 +1,22 @@
 #!/bin/bash
-pushd collector/cmd/dcos-metrics-collector
-make
-popd
+set -e
+export CGO_ENABLED=0
 
+COMPONENT="$1"
+PLATFORM=$(uname | tr [:upper:] [:lower:])
+GIT_REF=$(git describe --always)
+SOURCE_DIR=$(git rev-parse --show-toplevel)
+BUILD_DIR="${SOURCE_DIR}/build/${COMPONENT}"
+
+# Bring in dependencies required by schema/go/generator.go
+go get -u github.com/antonholmquist/jason
+
+rm -rf $BUILD_DIR && mkdir -p $BUILD_DIR
+cd "${SOURCE_DIR}/${COMPONENT}"
+
+# build metrics-schema package
+go generate
+
+# build binary
+cd cmd/dcos-metrics-collector
+go build -a -ldflags "-w" -o ${BUILD_DIR}/dcos-metrics-collector-${GIT_REF} *.go
```

Comment:
It's true that I've never had to use GDB or other debuggers on my built executables, but this may come in handy down the road. The trade off is a significantly larger binary (by percent, approx 25%) however, when we're talking about at most tens of megabytes, this doesn't really concern me, and I'd rather have an ability to debug until we have complete stability than not.